pkg/health/client/tree.go

```diff
@@ -0,0 +1,219 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package client
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+const (
+	indentSize = 3
+
+	link decoration = "│"
+	mid  decoration = "├──"
+	end  decoration = "└──"
+)
+
+type (
+	decoration string
+
+	val  interface{}
+	meta interface{}
+	tree interface {
+		addNode(v val) tree
+		addNodeWithMeta(v val, m meta) tree
+		addBranch(v val) tree
+		addBranchWithMeta(v val, meta meta) tree
+		find(val val) tree
+		String() string
+		asBytes() []byte
+	}
```

Comment:
I don't see much value in defining these interfaces.
This tree implementation seems something specialized for the health reporter, so I'd say we can just use `string` for `val` and `meta` and simply remove the `tree` interface. This should also simplify the code a bit (e.g.: no need to use `reflect.DeepEqual` in `find` and no need for the unused `addNode`).
WDYT?

Reply:
Good point. Thank you Fabio! Was thinking this could be useful somewhere else down the line but who knows?? I'll dumb it down.